tests/conftest.py

```diff
@@ -256,7 +257,8 @@ def test_assets_directory() -> pathlib.Path:
 @pytest.fixture(scope="module")
 def grpc_channel():
     """Gets the gRPC channel."""
-    with grpc.insecure_channel("localhost:31763") as channel:
+    with GrpcServerProcess() as proc:
+        channel = grpc.insecure_channel(f"localhost:{proc.server_port}")
         yield channel
```

Comment:
Now this creates two closable objects, but it only closes one.

Please create a separate `grpc_server_process` fixture. It should have scope=session, use a `with` block to close the proc, and `yield` the proc.

Change `grpc_channel` to depend on `grpc_server_process`. Restore the `with` block to the way it was before, which closes the channel.

Reply:
created a `grpc_server_process` fixture as per suggestion.